habitat/datasets/rearrange/rearrange_generator.py

```diff
@@ -423,8 +431,11 @@ def generate_single_episode(self) -> Optional[RearrangeEpisode]:
         ep_scene_handle = self.generate_scene()
 
         # Get the unique open receptacle
-        sampler = list(self._obj_samplers.values())[0]
-        target_receptacle = sampler.sample_receptacle(self.sim)
+        single_target_in_open_receptacle = self.cfg.unique_target
+        target_receptacle = None
+        if single_target_in_open_receptacle:
+            sampler = list(self._obj_samplers.values())[0]
```

Comment:
```suggestion
            sampler = next(iter(self._obj_samplers.values()))
```
Guaranteed to be constant